server/resp/src/main/java/org/infinispan/server/resp/commands/ArgumentUtils.java

```diff
@@ -39,14 +39,24 @@ public static double toDouble(byte[] argument, int offset) {
       if (argument == null || argument.length == 0)
          throw new NumberFormatException("Empty argument");
 
-      return Double.parseDouble(new String(argument, offset, argument.length - offset, CharsetUtil.US_ASCII));
+      return toDoubleWithInfinity(new String(argument, offset, argument.length - offset, CharsetUtil.US_ASCII));
    }
 
    public static double toDouble(byte[] argument) {
       if (argument == null || argument.length == 0)
          throw new NumberFormatException("Empty argument");
+      String sArg = toNumberString(argument);
+      return toDoubleWithInfinity(sArg);
+   }
 
-      return Double.parseDouble(toNumberString(argument));
+   private static double toDoubleWithInfinity(String sArg) {
```

Comment:
Created https://issues.redhat.com/browse/ISPN-16035 to update in the future. We can leave it to a follow-up since it is a specific case. Most of the time we would transform to String anyway to use the Double.parseDouble.